From 22ad7831036205f5dcd8f7fb0451f13e91c41c16 Mon Sep 17 00:00:00 2001
From: bbhtt <bbhtt.zn0i8@slmail.me>
Date: Mon, 1 Jul 2024 07:59:04 +0530
Subject: [PATCH] builder-module: Don't set libdir for autotools

Some configure scripts like perl's configure.gnu does not understand
libdir [1], but uses non-standard libpath. Passing libdir
unconditionally means it will fail

[1]: https://github.com/Perl/perl5/blob/8a9c915b0c7a3b732776db7e95336fc3ac1d37d6/configure.gnu#L88-L90
---
 src/builder-module.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/builder-module.c b/src/builder-module.c
index 7816fab7..910cd551 100644
--- a/src/builder-module.c
+++ b/src/builder-module.c
@@ -1797,7 +1797,7 @@ builder_module_build_helper (BuilderModule  *self,
           g_ptr_array_add (configure_args_arr, g_strdup_printf ("PREFIX='%s'", prefix));
           /* TODO: What parameter for qmake? */
         }
-      else /* autotools and meson */
+      else if (meson)
         {
           g_ptr_array_add (configure_args_arr, g_strdup_printf ("--prefix=%s", prefix));
           if (libdir)
@@ -1805,6 +1805,12 @@ builder_module_build_helper (BuilderModule  *self,
           else
             g_ptr_array_add (configure_args_arr, g_strdup_printf ("--libdir=%s/lib", prefix));
         }
+      else /* autotools */
+        {
+          g_ptr_array_add (configure_args_arr, g_strdup_printf ("--prefix=%s", prefix));
+          if (libdir)
+            g_ptr_array_add (configure_args_arr, g_strdup_printf ("--libdir=%s", libdir));
+        }
 
       g_ptr_array_add (configure_args_arr, configure_final_arg);
       g_ptr_array_add (configure_args_arr, NULL);
-- 
2.45.2

